File: backend/src/main/java/com/daniel/martins/youtubeclone/dto/VideoDto.java
package com.daniel.martins.youtubeclone.dto;

import com.daniel.martins.youtubeclone.model.Comment;
import com.daniel.martins.youtubeclone.model.VideoStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

import java.util.List;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VideoDto {

    private String id;
    private String sessionId;
    private String title;
    private String description;
    private String userId;
    private Set<String> tags;
    private String videoUrl;
    private VideoStatus videoStatus;
    private String thumbnailUrl;
    private Integer likeCount;
    private Integer dislikeCount;
    private Integer viewCount;

}

File: frontend/youtube-clone-ui/src/app/video-dto.ts
export interface VideoDto {
    id : string;
    sessionId : string;
    title : string ;
    description : string;
    tags : Array<string>;
    videoUrl : string;
    videoStatus : string;
    thumbnailUrl : string;
    likeCount: number;
    dislikeCount: number; 
    viewCount: number;
    userId : string;
    createdAt: string;
}